### PR TITLE
ebpf: split code

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [ ] Git log contains summary of the change.
- [ ] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Copy and paste the git log here.

Fixes: #2359 
Close: #505

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [x] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
@yanivagman what are your objections here ? I don't have much, as it is way better than currently what we have.

The cons are:

- will have to rebase the PRs we have opened
- will need indexing (cscope, tags) configured in your editor

The pros are:

- much faster parsing speed in any editor
- way more organized code
- initial work for future plugin mechanism
- rebases, although in "cons" will be straightforward (no code changes)

So for me this seems like a +1 (but maybe only after the PR from @geyslan is done, which creates the scopes and is a core for this milestone).

**Review by rafaeldtinoco — COMMENTED** [review]
needs discussion (more of a decision than a technical thing)

**Comment by yanivagman** [comment]
1. This should be merged only after #2537 which does changes all over the code
2. I think all of the field extractors (functions that take a kernel struct and return one of its fields) should be grouped into one file - overwise it's just too many files

**Comment by NDStrahilevitz** [comment]
> 1. This should be merged only after [Store event data in a map #2537](https://github.com/aquasecurity/tracee/pull/2537) which does changes all over the code
> 2. I think all of the field extractors (functions that take a kernel struct and return one of its fields) should be grouped into one file - overwise it's just too many files

1. Fine by me.
2. I disagree, this makes the helpers more focused and easier to find by "theme" from a glance at the filesystem, and allows for easier composition of helpers later in the same file (take a look at `network.c` for a good example of this in action). It's also useful later for plugin writers to be able to consume only the helpers they need. 
However I also know that this will mainly be a convenience concern for now so I believe we should take in concern the opinion of the other devs since they will know better what will be more convenient for them.

**Comment by AlonZivony** [comment]
@NDStrahilevitz, is there a reason for not using both header and c files?
The problem with ordering functions inside each header file will remain a problem with the current structure.

File: pkg/ebpf/c/common/arch.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#ifndef __TRACEE_TASK_H__
```
[the remark this answers is not in the document]
Comment (NDStrahilevitz) [reply pkg/ebpf/c/common/arch.h:1]: @yanivagman notion mentioned to use `#ifndef` iirc, any reason not to use `#pragma once`? I'm ok with both.

**Comment by NDStrahilevitz** [comment]
@yanivagman @rafaeldtinoco Since we've merged #2537 are we gtg with this one?

**Comment by geyslan** [comment]
@NDStrahilevitz As soon as I finish the multiscope with Rafael (some network code being put and removed) I'll be returning to this. 👍🏼

**Comment by rafaeldtinoco** [comment]
> @yanivagman @rafaeldtinoco Since we've merged #2537 are we gtg with this one?

I believe we should only merge this after the https://github.com/aquasecurity/tracee/pull/2238 (can be right after, but that one has already suffered numerous rebases, more than 2 hands can count =D).

**Comment by geyslan** [comment]
> > @yanivagman @rafaeldtinoco Since we've merged #2537 are we gtg with this one?
> 
> I believe we should only merge this after the #2238 (can be right after, but that one has already suffered numerous rebases, more than 2 hands can count =D).



#2238 was awarded.

File: pkg/ebpf/c/common/arch.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (yanivagman) [reply pkg/ebpf/c/common/arch.h:1]: I'm ok with both as well

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -4553,6 +1329,15 @@ static __always_inline void *get_trace_probe_from_trace_event_call(struct trace_
...
 
+enum perf_type_e
+{
+    PERF_TRACEPOINT,
+    PERF_KPROBE,
+    PERF_KRETPROBE,
+    PERF_UPROBE,
+    PERF_URETPROBE
+};
+
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:1340]: Why adding this?

Comment (NDStrahilevitz) [reply pkg/ebpf/c/tracee.bpf.c:1340]: Guess it's a leftover, i'll remove.

File: pkg/ebpf/c/common/syscalls.h [hunk 3]
```diff
@@ -0,0 +1,359 @@
+#ifndef __TRACEE_SYSCALLS_H__
+#define __TRACEE_SYSCALLS_H__
+
+#include <bpf/bpf_helpers.h>
+#include <types.h>
+
+#define UNDEFINED_SYSCALL 1000
+#define NO_SYSCALL        -1
+
+#if defined(bpf_target_x86)
```
[the remark this answers is not in the document]
Comment (NDStrahilevitz) [reply pkg/ebpf/c/common/syscalls.h:10]: Sure, we can merge them.

**Review by yanivagman — APPROVED** [review]
Thanks @NDStrahilevitz 
LGTM

**Pull request merged**